src/silo/preprocessing/sequence_info.cpp

```diff
@@ -0,0 +1,178 @@
+      );
+   }
+
+   return "list_string_agg(flatten([" + boost::join(list_transforms, ",") + "]))";
+}
+
+std::string SequenceInfo::getAAInsertionSelect() {
+   if (aa_sequence_names.empty()) {
+      return "''";
+   }
+   if (aa_sequence_names.size() == 1) {
+      return fmt::format(
+         "list_string_agg(list_transform(aminoAcidInsertions.{0}, x ->'{0}:' || x))",
+         aa_sequence_names.at(0)
+      );
+   }
+
+   std::vector<std::string> list_transforms;
+   for (const std::string& sequence_name : aa_sequence_names) {
+      list_transforms.push_back(
+         fmt::format("list_transform(aminoAcidInsertions.{0}, x ->'{0}:' || x)", sequence_name)
+      );
+   }
+
+   return "list_string_agg(flatten([" + boost::join(list_transforms, ",") + "]))";
+}
+
+void SequenceInfo::validate(duckdb::Connection& connection, std::string_view input_filename) const {
+   auto result = connection.Query(fmt::format(
+      "SELECT json_keys(alignedNucleotideSequences), json_keys(alignedAminoAcidSequences) "
+      "FROM "
+      "'{}' LIMIT 1; ",
+      input_filename
+   ));
+   if (result->HasError()) {
+      throw silo::preprocessing::PreprocessingException(
+         "Preprocessing exception when retrieving the fields 'alignedNucleotideSequences' and "
+         "'alignedAminoAcidSequences', duckdb threw with error: " +
+         result->GetError()
+      );
+   }
+   if (result->RowCount() == 0) {
+      throw silo::preprocessing::PreprocessingException(
+         fmt::format("File {} is empty, which must not be empty at this point", input_filename)
+      );
+   }
+   if (result->RowCount() > 1) {
+      throw silo::preprocessing::PreprocessingException(
+         "Internal exception, expected Row Count=1, actual " + std::to_string(result->RowCount())
+      );
+   }
+
+   auto nuc_sequence_names_to_validate = extractStringListValue(*result, 0, 0);
+   auto aa_sequence_names_to_validate = extractStringListValue(*result, 0, 1);
+
+   for (const std::string& name : nuc_sequence_names_to_validate) {
+      if (std::find(nuc_sequence_names.begin(), nuc_sequence_names.end(), name) == nuc_sequence_names.end()) {
+         throw silo::preprocessing::PreprocessingException(fmt::format(
+            "The aligned nucleotide sequence {} which is contained in the input file {} is "
+            "not contained in the reference sequences.",
+            name,
+            input_filename
+         ));
+      }
+   }
+   for (const std::string& name : nuc_sequence_names) {
+      if (std::find(nuc_sequence_names_to_validate.begin(), nuc_sequence_names_to_validate.end(), name)
+          == nuc_sequence_names_to_validate.end()) {
+         throw silo::preprocessing::PreprocessingException(fmt::format(
+            "The aligned nucleotide sequence {} which is contained in the reference sequences is "
+            "not contained in the input file {}.",
+            name,
+            input_filename
+         ));
+      }
+   }
```

Comment:
In the Pathoplexus backend we allow that a sequence entry only has some segments. We don't require that it has all segments. From what I understood it seems to be a normal case that the sequenced result only contains some segments.